api/src/services/import-services/critter/critter-header-configs.ts

```diff
@@ -0,0 +1,183 @@
+ *
+ * @param {Object} rowDictionary The row dictionary.
+ * @param {CSVConfigUtils<CritterCSVConfig>} configUtils The CSV config utils.
+ * @returns {*} {(params: CSVParams) => string | undefined} The set cell value callback
+ */
+export const getCritterCollectionUnitCellSetter = (
+  rowDictionary: { [tsn: number]: { [header: string]: { [unit: string]: string } } },
+  configUtils: CSVConfigUtils<CritterCSVConfig>
+): ((params: CSVParams) => string | undefined) => {
+  return (params: CSVParams) => {
+    if (!params.cell) {
+      return undefined;
+    }
+
+    const tsn = Number(configUtils.getCellValue('ITIS_TSN', params.row));
+    const unit = String(params.cell).toLowerCase();
+
+    return get(rowDictionary, [tsn, params.header, unit]);
+  };
+};
+
+/**
+ * Get the critter sex cell validator.
+ *
+ * @param {Object} rowDictionary The row dictionary.
+ * @param {CSVConfigUtils<CritterCSVConfig>} configUtils The CSV config utils.
+ * @returns {*} {(params: CSVParams) => CSVError[]} The validate cell callback
+ */
+export const getCritterSexCellValidator = (
+  rowDictionary: { [tsn: number]: { [sex: string]: string } },
+  configUtils: CSVConfigUtils<CritterCSVConfig>
+): ((params: CSVParams) => CSVError[]) => {
+  return (params: CSVParams) => {
+    const cellErrors = validateZodCell(params, z.string().trim().min(1).max(50));
+
+    if (cellErrors.length) {
+      return cellErrors;
+    }
+
+    const rowTsn = Number(configUtils.getCellValue('ITIS_TSN', params.row));
+    const cellValue = String(params.cell).toLowerCase();
+
+    const rowDictionaryTsn = get(rowDictionary, rowTsn);
+    const rowDictionarySex = get(rowDictionary, [rowTsn, cellValue]);
+
+    if (!rowDictionaryTsn) {
+      cellErrors.push({
+        error: `Sex is not a supported attribute for TSN: ${rowTsn}`,
+        solution: `Use a valid TSN that supports sex, or contact a system administrator to add additional sex values.`,
+        ...getCSVParamsError(params)
+      });
+    } else if (!rowDictionarySex) {
+      cellErrors.push({
+        error: `Sex cell value is invalid`,
+        solution: `Use valid sex option`,
+        values: Object.keys(rowDictionaryTsn),
+        ...getCSVParamsError(params)
+      });
+    }
+
+    return cellErrors;
+  };
+};
+
+/**
+ * Get the critter sex cell setter.
+ *
+ * @param {Object} rowDictionary The row dictionary.
+ * @param {CSVConfigUtils<CrittterCSVConfig>} configUtils The CSV config utils.
+ * @returns {*} {(params: CSVParams) => string} The validate cell callback
+ */
+export const getCritterSexCellSetter = (
+  rowDictionary: { [tsn: number]: { [sex: string]: string } },
+  configUtils: CSVConfigUtils<CritterCSVConfig>
+): ((params: CSVParams) => string) => {
+  return (params: CSVParams) => {
+    const tsn = Number(configUtils.getCellValue('ITIS_TSN', params.row));
+    const cellValue = String(params.cell).toLowerCase();
+
+    return get(rowDictionary, [tsn, cellValue]);
```

Comment:
Some extra comments or jsdoc might be helpful here. Its a bit hard to understand whats happening here, especially because the `tsn` const is calling `getCellValue` but the other const is called `cellValue`. And the lodash function doesnt include any jsdoc on hover.

Reply:
I updated the jsdoc and renamed some variables for clarity.

Note: The rowDictionary was my best attempt at handling these types of validation relationships, but I'm open to additional solutions if theres a better data structure be it native or custom that I didn't think of